cis194/week7/tien/src/JoinListBuffer.hs

```diff
@@ -0,0 +1,36 @@
+{-# OPTIONS_GHC -Wall #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+
+module JoinListBuffer where
+
+import JoinList
+  ( JoinList(Append, Empty, Single)
+  , (+++)
+  , dropJ
+  , indexJ
+  , tag
+  , takeJ
+  )
+import Scrabble
+import Sized
+
+import Buffer
+
+-- exercise 4 --
+instance Buffer (JoinList (Score, Size) String) where
+  toString (Single _ x) = x
```

Comment:
You've implemented `jlToList` so you don't need to pattern match one more time here.

```suggestion
  toString = unlines . jlToList
```